packages/browser-data-collector/src/index.ts

```diff
@@ -4,7 +4,7 @@
 import { ReportImpl } from './report';
 import { send } from './transports/logstash';
 
-const inFlightReporters: Map< string, ReportImpl > = new Map();
+const inFlightReporters: Map< string, Promise< ReportImpl > > = new Map();
```

Comment:
Big internal change here. Now reports starts asynchronously, so it is theoretically possible to try to `stop` a report that is still `start`ing.

This is done to keep the `collector` type consistent (i.e. start collectors can be async too, like stop collectors)

The reason is to avoid having a weird state where a report doesn't exist in `inFlightReporters`, we initialize it but has to wait for it to start` and meanwhile another report comes in with the same id. It could cause to have two "starting" reports for the same id, which will get messy.